### PR TITLE
Check that all scannable items point to different pdf

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-704

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java [hunk 1]
```diff
@@ -86,21 +88,38 @@ public static void assertEnvelopeContainsOcrDataIfRequired(InputEnvelope envelop
...
+                .map(it -> it.fileName)
+                .collect(groupingBy(it -> it, counting()))
+                .entrySet()
+                .stream()
+                .filter(it -> it.getValue() > 1)
+                .map(it -> it.getKey())
+                .collect(toList());
+        
+        if (!duplicateFileNames.isEmpty()) {
+            problems.add("Duplicate scanned items file names: " + String.join(", ", duplicateFileNames));
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: Perhaps clearer if notified where exactly? Could even get the `zipFileName` there if needed

```suggestion
            problems.add("Duplicate scanned items in metafile: " + String.join(", ", duplicateFileNames));
```

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java [hunk 2]
```diff
@@ -86,21 +88,38 @@ public static void assertEnvelopeContainsOcrDataIfRequired(InputEnvelope envelop
...
+            envelope
+                .scannableItems
+                .stream()
+                .map(it -> it.fileName)
+                .collect(groupingBy(it -> it, counting()))
+                .entrySet()
+                .stream()
+                .filter(it -> it.getValue() > 1)
+                .map(it -> it.getKey())
+                .collect(toList());
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: I've been looking into this for a while but only coming with either nasty hack or classic foreach loop with 2 sets figuring out what is duplicate what is not.

This streaming put me off.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java [hunk 3]
```diff
@@ -86,21 +88,38 @@ public static void assertEnvelopeContainsOcrDataIfRequired(InputEnvelope envelop
      * @param pdfs     to assert against
      */
     public static void assertEnvelopeHasPdfs(InputEnvelope envelope, List<Pdf> pdfs) {
+        List<String> problems = new ArrayList<>();
+
+        List<String> duplicateFileNames =
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: Let's extract this to its separate method maybe

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: Currently existing errors are: 
>Not declared PDFs: ...
and
>Missing PDFs:...

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: Based on exception type and appropriate error code is set on the notification.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: Yea thought so. Still makes clear duplicate names **are** in metafile itself. Don't mind hence approved

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java]: done

**Pull request merged**